### PR TITLE
Refactor URI overrides

### LINKED ISSUE #10005: [Issue]: Docker: JELLYFIN_PublishedServerUrl gets ignored
### Please describe your bug

It seems the environment variable ```JELLYFIN_PublishedServerUrl``` as [shown in the docker-compose documentation](https://jellyfin.org/docs/general/installation/container#using-docker-compose) gets ignored for SSDP multicasts in current versions of JellyFin. The NOTIFY messages published by the container always contain the host's IP address instead of the address passed using that environment variable.

I found [this old issue](https://github.com/jellyfin/jellyfin/issues/5280) describing the same problem that was solved back then. But it seems a lot has changed in [DlnaEntryPoint.cs](https://github.com/jellyfin/jellyfin/blob/master/Emby.Dlna/Main/DlnaEntryPoint.cs) since then. I tracked down the two versions where it stopped working: Works fine in [10.8.0 Alpha 1](https://github.com/jellyfin/jellyfin/releases/tag/v10.8.0-alpha1), stopped working in [10.8.0 Alpha 2](https://github.com/jellyfin/jellyfin/releases/tag/v10.8.0-alpha2).
I haven't tested it, but I guess this is not only a problem with the docker environment variable, but wouldn't work with the [command line argument](https://jellyfin.org/docs/general/administration/configuration#main-configuration-options) either.

## Steps to reproduce
  1. Open Wireshark on a machine on the same network and filter for ```ssdp && ip.src == <dockerHostAddress>```
  2. Launch JellyFin in a docker container with version ```10.8.10``` with ```JELLYFIN_PublishedServerUrl``` set
  ```bash
  docker run -d \
 --name jellyfin \
 --user 1000:1000 \
 --net=host \
 --volume jellyfin-config:/config \
 --volume jellyfin-cache:/cache \
 --restart=unless-stopped \
 --env JELLYFIN_PublishedServerUrl=http://example.com \
 jellyfin/jellyfin:10.8.10
  ```
  3. Make sure to check ```Enable DLNA Server``` option in ```Admin Bashboard -> DLNA```
  4. Find the ```SSDP NOTIFY``` message in wireshark with the wrong ```LOCATION``` attribute
  ```
Simple Service Discovery Protocol
    NOTIFY * HTTP/1.1\r\n
    HOST: 239.255.255.250:1900\r\n
    DATE: Sun, 16 Jul 2023 09:04:38 GMT\r\n
    CACHE-CONTROL: max-age = 1800\r\n
    LOCATION: http://192.168.1.5:8096/dlna/a6f2204e-538d-4fa0-b059-d517e92069a1/description.xml\r\n
    SERVER: Linux/Unix 5.15.107.0 UPnP/1.0 RSSDP/1.0\r\n
    NTS: ssdp:alive\r\n
    NT: upnp:rootdevice\r\n
    USN: uuid:a6f2204e-538d-4fa0-b059-d517e92069a1::upnp:rootdevice\r\n
    \r\n
    [Full request URI: http://239.255.255.250:1900*]
```
  5. Stop and remove the container
  ```bash
  docker stop jellyfin
  docker remove jellyfin
  ```
  6. Start the container using image version ```10.8.0-alpha1```
  ```bash
  docker run -d \
 --name jellyfin \
 --user 1000:1000 \
 --net=host \
 --volume jellyfin-config:/config \
 --volume jellyfin-cache:/cache \
 --restart=unless-stopped \
 --env JELLYFIN_PublishedServerUrl=http://example.com \
 jellyfin/jellyfin:10.8.0-alpha1
```
  7. Find the ```SSDP NOTIFY``` message in wireshark with the correct ```LOCATION``` attribute
  ```
Simple Service Discovery Protocol
    NOTIFY * HTTP/1.1\r\n
    HOST: 239.255.255.250:1900\r\n
    DATE: Sun, 16 Jul 2023 09:06:37 GMT\r\n
    CACHE-CONTROL: max-age = 1800\r\n
    LOCATION: http://example.com:8096/dlna/a6f2204e538d4fa0b059d517e92069a1/description.xml\r\n
    SERVER: Linux/Unix 5.15.107.0 UPnP/1.0 RSSDP/1.0\r\n
    NTS: ssdp:alive\r\n
    NT: urn:schemas-upnp-org:service:ContentDirectory:1\r\n
    USN: uuid:a6f2204e538d4fa0b059d517e92069a1::urn:schemas-upnp-org:service:ContentDirectory:1\r\n
    \r\n
    [Full request URI: http://239.255.255.250:1900*]
  ```
  However, for use with a reverse proxy JellyFin should not automatically add the default port ```8096``` to the ```PublishedServerUrl``` option but assume that everything needed is already included in that value.

### Jellyfin Version

10.8.z

### if other:

_No response_

### Environment

```markdown
- OS: Debian 12 Bookworm
- Linux Kernel: 6.1
- Virtualization: Docker 24.0.4
- Clients: Android
- Browser: Firefox  
- FFmpeg Version: -
- Playback Method: DLNA
- Hardware Acceleration: Intel Quicksync
- GPU Model: Intel UHD770
- Plugins: none
- Reverse Proxy: traefik v2
- Base URL: none
- Networking: Host
- Storage: local
```

### Jellyfin logs

_No response_

### FFmpeg logs

_No response_

### Please attach any browser or client logs here

_No response_

### Please attach any screenshots here

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #10070: [Issue]: Jellyfin unstable not binding to localhost
### Please describe your bug

Jellyfin stable release binds to localhost as expected.

Jellyfin unstable releases no longer bind to localhost, I believe as a result of this PR: https://github.com/jellyfin/jellyfin/pull/8147

When jellyfin is run inside of a container for instance, `netstat -tulpn | grep LISTEN` output is as follows:
Stable:
`tcp        0      0 0.0.0.0:8096            0.0.0.0:*               LISTEN      -`
Unstable:
`tcp        0      0 172.17.0.2:8096         0.0.0.0:*               LISTEN      -`
In this case, `172.17.0.2` is the container IP. Somehow jellyfin detects the container IP and only listens at that IP.

Relevant log snippet:
```
[20:45:45] [INF] [1] Jellyfin.Networking.Manager.NetworkManager: Defined LAN addresses: ["127.0.0.1/8", "10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16"]
[20:45:45] [INF] [1] Jellyfin.Networking.Manager.NetworkManager: Defined LAN exclusions: []
[20:45:45] [INF] [1] Jellyfin.Networking.Manager.NetworkManager: Using LAN addresses: ["127.0.0.1/8", "10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16"]
[20:45:45] [INF] [1] Jellyfin.Networking.Manager.NetworkManager: Using bind addresses: ["172.17.0.2"]
```

Perhaps this is intended behavior, but it causes other issues for things that rely on localhost binding, like health checks. The container IP is ephemeral.

Thanks

### Jellyfin Version

Other

### if other:

10.9

### Environment

```markdown
Fresh instance in an ubuntu jammy based docker container for reproducability purposes. Jellyfin started for the first time with no existing data.
```

### Jellyfin logs

```shell
[20:45:45] [INF] [1] Main: Jellyfin version: 10.9.0
[20:45:45] [INF] [1] Main: Environment Variables: ["[JELLYFIN_CONFIG_DIR, /config]", "[JELLYFIN_CACHE_DIR, /config/cache]", "[JELLYFIN_LOG_DIR, /config/log]", "[JELLYFIN_WEB_DIR, /usr/share/jellyfin/web]", "[JELLYFIN_DATA_DIR, /config/data]"]
[20:45:45] [INF] [1] Main: Arguments: ["/usr/lib/jellyfin/bin/jellyfin.dll", "--ffmpeg=/usr/lib/jellyfin-ffmpeg/ffmpeg"]
[20:45:45] [INF] [1] Main: Operating system: Linux 6.1.38-Unraid #2 SMP PREEMPT_DYNAMIC Mon Jul 10 09:50:25 PDT 2023
[20:45:45] [INF] [1] Main: Architecture: X64
[20:45:45] [INF] [1] Main: 64-Bit Process: True
[20:45:45] [INF] [1] Main: User Interactive: True
[20:45:45] [INF] [1] Main: Processor count: 12
[20:45:45] [INF] [1] Main: Program data path: /config/data
[20:45:45] [INF] [1] Main: Web resources path: /usr/share/jellyfin/web
[20:45:45] [INF] [1] Main: Application directory: /usr/lib/jellyfin/bin/
[20:45:45] [INF] [1] Jellyfin.Server.Migrations.MigrationRunner: Marking following migrations as applied because this is a fresh install: ["CreateNetworkConfiguration", "MigrateMusicBrainzTimeout", "MigrateNetworkConfiguration"]
[20:45:45] [INF] [1] Emby.Server.Implementations.AppBase.BaseConfigurationManager: Setting cache path: /config/cache
[20:45:45] [INF] [1] Emby.Server.Implementations.ApplicationHost: Loading assemblies
[20:45:45] [INF] [1] Jellyfin.Networking.Manager.NetworkManager: Defined LAN addresses: ["127.0.0.1/8", "10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16"]
[20:45:45] [INF] [1] Jellyfin.Networking.Manager.NetworkManager: Defined LAN exclusions: []
[20:45:45] [INF] [1] Jellyfin.Networking.Manager.NetworkManager: Using LAN addresses: ["127.0.0.1/8", "10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16"]
[20:45:45] [INF] [1] Jellyfin.Networking.Manager.NetworkManager: Using bind addresses: ["172.17.0.2"]
[20:45:45] [INF] [1] Emby.Server.Implementations.ApplicationHost: There are pending EFCore migrations in the database. Applying... (This may take a while, do not stop Jellyfin)
[20:45:46] [WRN] [6] Microsoft.EntityFrameworkCore.Migrations: An operation of type 'SqlOperation' will be attempted while a rebuild of table 'ImageInfos' is pending. The database may not be in an expected state. Review the SQL generated by this migration to help diagnose any failures. Consider moving these operations to a subsequent migration.
[20:45:46] [WRN] [6] Microsoft.EntityFrameworkCore.Migrations: An operation of type 'SqlOperation' will be attempted while a rebuild of table 'ImageInfos' is pending. The database may not be in an expected state. Review the SQL generated by this migration to help diagnose any failures. Consider moving these operations to a subsequent migration.
[20:45:46] [WRN] [6] Microsoft.EntityFrameworkCore.Migrations: An operation of type 'SqlOperation' will be attempted while a rebuild of table 'Permissions' is pending. The database may not be in an expected state. Review the SQL generated by this migration to help diagnose any failures. Consider moving these operations to a subsequent migration.
[20:45:46] [WRN] [6] Microsoft.EntityFrameworkCore.Migrations: An operation of type 'SqlOperation' will be attempted while a rebuild of table 'Permissions' is pending. The database may not be in an expected state. Review the SQL generated by this migration to help diagnose any failures. Consider moving these operations to a subsequent migration.
[20:45:46] [WRN] [6] Microsoft.EntityFrameworkCore.Migrations: An operation of type 'SqlOperation' will be attempted while a rebuild of table 'Preferences' is pending. The database may not be in an expected state. Review the SQL generated by this migration to help diagnose any failures. Consider moving these operations to a subsequent migration.
[20:45:46] [WRN] [6] Microsoft.EntityFrameworkCore.Migrations: An operation of type 'SqlOperation' will be attempted while a rebuild of table 'Preferences' is pending. The database may not be in an expected state. Review the SQL generated by this migration to help diagnose any failures. Consider moving these operations to a subsequent migration.
[20:45:46] [WRN] [6] Microsoft.EntityFrameworkCore.Migrations: An operation of type 'SqlOperation' will be attempted while a rebuild of table 'Users' is pending. The database may not be in an expected state. Review the SQL generated by this migration to help diagnose any failures. Consider moving these operations to a subsequent migration.
[20:45:46] [WRN] [6] Microsoft.EntityFrameworkCore.Migrations: An operation of type 'SqlOperation' will be attempted while a rebuild of table 'Users' is pending. The database may not be in an expected state. Review the SQL generated by this migration to help diagnose any failures. Consider moving these operations to a subsequent migration.
[20:45:46] [INF] [13] Emby.Server.Implementations.ApplicationHost: EFCore migrations applied successfully
[20:45:46] [INF] [13] Emby.Server.Implementations.AppBase.BaseConfigurationManager: Saving system configuration
[20:45:46] [INF] [13] Emby.Server.Implementations.AppBase.BaseConfigurationManager: Setting cache path: /config/cache
[20:45:46] [INF] [13] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: TMDb 10.9.0.0
[20:45:46] [INF] [13] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: Studio Images 10.9.0.0
[20:45:46] [INF] [13] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: OMDb 10.9.0.0
[20:45:46] [INF] [13] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: MusicBrainz 10.9.0.0
[20:45:46] [INF] [13] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: AudioDB 10.9.0.0
[20:45:46] [INF] [13] Jellyfin.Server.Migrations.MigrationRunner: Marking following migrations as applied because this is a fresh install: ["DisableTranscodingThrottling", "CreateLoggingConfigHeirarchy", "MigrateActivityLogDatabase", "RemoveDuplicateExtras", "MigrateUserDatabase", "MigrateDisplayPreferencesDatabase", "RemoveDownloadImagesInAdvance", "MigrateAuthenticationDatabase", "FixPlaylistOwner", "MigrateRatingLevels"]
[20:45:46] [INF] [13] Jellyfin.Server.Migrations.MigrationRunner: Applying migration 'AddDefaultPluginRepository'
[20:45:46] [INF] [13] Emby.Server.Implementations.AppBase.BaseConfigurationManager: Saving system configuration
[20:45:46] [INF] [13] Emby.Server.Implementations.AppBase.BaseConfigurationManager: Setting cache path: /config/cache
[20:45:46] [INF] [13] Jellyfin.Server.Migrations.MigrationRunner: Migration 'AddDefaultPluginRepository' applied successfully
[20:45:46] [INF] [13] Jellyfin.Server.Migrations.MigrationRunner: Applying migration 'ReaddDefaultPluginRepository'
[20:45:46] [INF] [13] Jellyfin.Server.Migrations.MigrationRunner: Migration 'ReaddDefaultPluginRepository' applied successfully
[20:45:46] [WRN] [13] Microsoft.AspNetCore.DataProtection.Repositories.EphemeralXmlRepository: Using an in-memory repository. Keys will not be persisted to storage.
[20:45:46] [WRN] [13] Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager: Neither user profile nor HKLM registry available. Using an ephemeral key repository. Protected data will be unavailable when application exits.
[20:45:46] [INF] [13] Main: Kestrel is listening on 172.17.0.2
[20:45:46] [WRN] [13] Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager: No XML encryptor configured. Key {a8614aec-1776-4daf-8f7e-c7dd85166148} may be persisted to storage in unencrypted form.
[20:45:46] [INF] [13] Emby.Server.Implementations.ApplicationHost: Running startup tasks
[20:45:46] [INF] [13] Emby.Server.Implementations.ScheduledTasks.TaskManager: Daily trigger for Extract Chapter Images set to fire at 2023-08-02 02:00:00.000 +00:00, which is 05:14:13.1673771 from now.
[20:45:46] [INF] [13] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Found ffmpeg version 5.1.3
[20:45:46] [INF] [13] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Available decoders: ["libdav1d", "av1", "av1_cuvid", "av1_qsv", "h264", "h264_qsv", "h264_cuvid", "hevc", "hevc_qsv", "hevc_cuvid", "mpeg2video", "mpeg2_qsv", "mpeg2_cuvid", "mpeg4", "mpeg4_cuvid", "msmpeg4", "vc1_qsv", "vc1_cuvid", "vp8", "libvpx", "vp8_cuvid", "vp8_qsv", "vp9", "libvpx-vp9", "vp9_cuvid", "vp9_qsv", "aac", "ac3", "dca", "flac", "mp3", "truehd"]
[20:45:46] [INF] [13] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Available encoders: ["libsvtav1", "libx264", "h264_amf", "h264_nvenc", "h264_qsv", "h264_v4l2m2m", "h264_vaapi", "libx265", "hevc_amf", "hevc_nvenc", "hevc_qsv", "hevc_vaapi", "mpeg4", "msmpeg4", "libvpx", "libvpx-vp9", "aac", "libfdk_aac", "ac3", "dca", "flac", "libmp3lame", "libopus", "truehd", "libvorbis", "srt"]
[20:45:46] [INF] [13] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Available filters: ["deinterlace_qsv", "deinterlace_vaapi", "hwupload_cuda", "hwupload_vaapi", "libplacebo", "overlay_opencl", "overlay_qsv", "overlay_vaapi", "overlay_vulkan", "overlay_cuda", "procamp_vaapi", "scale_cuda", "scale_opencl", "scale_qsv", "scale_vaapi", "scale_vulkan", "tonemap_cuda", "tonemap_opencl", "tonemap_vaapi", "vpp_qsv", "yadif_cuda", "zscale", "alphasrc"]
[20:45:47] [INF] [13] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Available hwaccel types: ["cuda", "vaapi", "qsv", "drm", "opencl", "vulkan"]
[20:45:47] [INF] [13] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: FFmpeg: /usr/lib/jellyfin-ffmpeg/ffmpeg
[20:45:47] [INF] [13] Emby.Server.Implementations.ApplicationHost: ServerId: 930cd2c9fe214871a568eb21e9844969
[20:45:47] [INF] [6] Emby.Server.Implementations.ApplicationHost: Executed all pre-startup entry points in 0:00:00.0633776
[20:45:47] [INF] [6] Emby.Server.Implementations.ApplicationHost: Core startup complete
[20:45:47] [INF] [6] Emby.Server.Implementations.ApplicationHost: Executed all post-startup entry points in 0:00:00.1277107
[20:45:47] [INF] [6] Main: Startup complete 0:00:02.3759644
[20:45:49] [INF] [6] Emby.Server.Implementations.ScheduledTasks.TaskManager: TaskCleanCollections Completed after 0 minute(s) and 0 seconds
[20:45:50] [INF] [11] Emby.Server.Implementations.ScheduledTasks.TaskManager: Update Plugins Completed after 0 minute(s) and 0 seconds
```

### FFmpeg logs

```shell
N/A
```

### Please attach any browser or client logs here

N/A

### Please attach any screenshots here

Screenshot 2023-08-02 102750
While the settings menu has this blurb, it no longer seems to be accurate as it's set to `blank` by default but it does not bind to all available addresses.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Shadowghost**
Refactor how URI overrides are loaded and created. It should now respect `JELLYFIN_PublishedServerUrl` again.
The override set via `JELLYFIN_PublishedServerUrl` supercedes any other override set via the UI or config file (and overrides them).

**Changes**
* Fix `GetAllBindInterfaces` -> fixes Kestrel binding on startup
* Do not exclude interfaces from unicast binding if the interface is not multicast capable
* Properly handle multicast binding for Linux, Windows and macOS
* Exlcude localhost manually from multicast on macOS due to https://github.com/dotnet/runtime/issues/24340
* Refactor URI override handling
* Respect `JELLYFIN_PublishedServerUrl`
* Introduce new class `PublishedServerUriOverride`
* Subnet parser now properly creates the catchall subnets if `0.0.0.0` and `::0` are input without subnet mask
* Cleanup NetworkManager logging and only show network information on startup (on network change printing is now debug-only)

**Issues**
* Fixes #10005
* Fixes #10070

### CODE REVIEW

**Comment by Shadowghost** [comment]
I don't think so. But you can test it without docker too (I'll do it myself if I find the time...)

**Comment by PrplHaz4** [comment]
> Is there a good way to generate a docker image from this PR for testing?
> 
> I don't have a build env setup to build Jellyfin, and we need to test it with Docker anyways...

These instructions served me well in the past to test without standing up the whole jf toolchain:
https://jellyfin.org/docs/general/contributing/development#building-and-testing-inside-a-docker-container

File: Emby.Server.Implementations/ApplicationHost.cs [hunk 1]
```diff
@@ -166,6 +166,8 @@ public abstract class ApplicationHost : IServerApplicationHost, IAsyncDisposable
 
             _xmlSerializer = new MyXmlSerializer();
             ConfigurationManager = new ServerConfigurationManager(ApplicationPaths, LoggerFactory, _xmlSerializer, _fileSystemManager);
+            ConfigurationManager.RegisterConfiguration<NetworkConfigurationFactory>();
+            NetManager = new NetworkManager(ConfigurationManager, startupConfig, LoggerFactory.CreateLogger<NetworkManager>());
```
Comment (cvium) [inline Emby.Server.Implementations/ApplicationHost.cs:170]: Why are you creating it twice?

**Comment by Bond-009** [comment]
Failing test on windows
```
[xUnit.net 00:00:00.57]     Jellyfin.Networking.Tests.NetworkExtensionsTests.TryParse_ValidHostStrings_True(address: "www.google.co.uk") [FAIL]
  Failed Jellyfin.Networking.Tests.NetworkExtensionsTests.TryParse_ValidHostStrings_True(address: "www.google.co.uk") [149 ms]
  Error Message:
   Assert.True() Failure
Expected: True
Actual:   False
  Stack Trace:
     at Jellyfin.Networking.Tests.NetworkExtensionsTests.TryParse_ValidHostStrings_True(String address) in D:\a\1\s\tests\Jellyfin.Networking.Tests\NetworkExtensionsTests.cs:line 30
   at InvokeStub_NetworkExtensionsTests.TryParse_ValidHostStrings_True(Object, Object, IntPtr*)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
Results File: D:\a\_temp\VssAdministrator_fv-az644-602_2023-08-15_01_15_02.trx

Failed!  - Failed:     1, Passed:   122, Skipped:     0, Total:   123, Duration: 1 s - Jellyfin.Networking.Tests.dll (net7.0)
```

**Comment by Shadowghost** [comment]
Failing tests are caused by a DNS resolution error when resolving `www.google.co.uk`. Not sure how to handle this, they are running fine locally.

**Comment by cvium** [comment]
> Failing tests are caused by a DNS resolution error when resolving `www.google.co.uk`. Not sure how to handle this, they are running fine locally.

We shouldn't have non-integration test making external calls

File: MediaBrowser.Model/Net/PublishedServerUriOverride.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+public class PublishedServerUriOverride
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PublishedServerUriOverride"/> class.
+    /// </summary>
+    /// <param name="data">The <see cref="IPData"/>.</param>
+    /// <param name="overrideUri">The override.</param>
+    /// <param name="internalOverride">A value indicating whether the override is for internal requests.</param>
+    /// <param name="externalOverride">A value indicating whether the override is for external requests.</param>
+    public PublishedServerUriOverride(IPData data, string overrideUri, bool internalOverride = true, bool externalOverride = true)
```
Comment (cvium) [inline MediaBrowser.Model/Net/PublishedServerUriOverride.cs:15]: Having ctor _and_ public setters seems a bit much

File: Jellyfin.Networking/Manager/NetworkManager.cs [hunk 3]
```diff
@@ -892,31 +940,34 @@ private bool MatchesPublishedServerUrl(IPAddress source, bool isInExternalSubnet
...
             int? port = null;
 
-            var validPublishedServerUrls = _publishedServerUrls.Where(x => x.Key.Address.Equals(IPAddress.Any)
-                                                || x.Key.Address.Equals(IPAddress.IPv6Any)
-                                                || x.Key.Subnet.Contains(source))
-                                            .DistinctBy(x => x.Key)
-                                            .OrderBy(x => x.Key.Address.Equals(IPAddress.Any)
-                                                || x.Key.Address.Equals(IPAddress.IPv6Any))
+            // Only consider subnets including the source IP, prefering specific overrides
+            var validPublishedServerUrls = new List<PublishedServerUriOverride>();
```
Comment (cvium) [inline Jellyfin.Networking/Manager/NetworkManager.cs:944]: isn't this empty list unused?

Comment (pitbuster) [reply Jellyfin.Networking/Manager/NetworkManager.cs:944]: the list seems to be filled-in in the if-else below and the used in the foreach afterwards :)

Comment (cvium) [reply Jellyfin.Networking/Manager/NetworkManager.cs:944]: It is not filled in, it's re-assigned

**Comment by satmandu** [comment]
@Shadowghost Any luck on getting this merged?

**Comment by Shadowghost** [comment]
I still want to fix a bug concerning multicast/broadcast binding on Windows. But it is a bit annoying to debug and I don't have that much time right now...

**Comment by satmandu** [comment]
@Shadowghost Any chance of leaving that Windows issue to a later PR, since this already fixes two issues?  (I'm currently building your branch rebased against master, but I would love to not have to do that.)

**Comment by satmandu** [comment]
I'm having problems rebasing this to master...

**Comment by satmandu** [comment]
Rather, on attempted rebasing I'm getting this when attempting to build a docker image:

```
115.2   Jellyfin.Drawing.Skia -> /repo/src/Jellyfin.Drawing.Skia/bin/Release/net7.0/Jellyfin.Drawing.Skia.dll           
117.4   Emby.Photos -> /repo/Emby.Photos/bin/Release/net7.0/Emby.Photos.dll                                             
122.1   MediaBrowser.LocalMetadata -> /repo/MediaBrowser.LocalMetadata/bin/Release/net7.0/MediaBrowser.LocalMetadata.dll
122.4   MediaBrowser.XbmcMetadata -> /repo/MediaBrowser.XbmcMetadata/bin/Release/net7.0/MediaBrowser.XbmcMetadata.dll   
126.7   Jellyfin.Server.Implementations -> /repo/Jellyfin.Server.Implementations/bin/Release/net7.0/Jellyfin.Server.Implementations.dll                                                                                                         
126.7   Jellyfin.Networking -> /repo/Jellyfin.Networking/bin/Release/net7.0/Jellyfin.Networking.dll                     
128.0   RSSDP -> /repo/RSSDP/bin/Release/net7.0/RSSDP.dll                                                               
128.1   MediaBrowser.Providers -> /repo/MediaBrowser.Providers/bin/Release/net7.0/MediaBrowser.Providers.dll            
128.1   Jellyfin.Drawing -> /repo/src/Jellyfin.Drawing/bin/Release/net7.0/Jellyfin.Drawing.dll                          
132.4   MediaBrowser.MediaEncoding -> /repo/MediaBrowser.MediaEncoding/bin/Release/net7.0/MediaBrowser.MediaEncoding.dll132.5   Emby.Dlna -> /repo/Emby.Dlna/bin/Release/net7.0/Emby.Dlna.dll                                                   
133.7   Jellyfin.MediaEncoding.Hls -> /repo/src/Jellyfin.MediaEncoding.Hls/bin/Release/net7.0/Jellyfin.MediaEncoding.Hls.dll                                                                                                                    
142.2   Jellyfin.Api -> /repo/Jellyfin.Api/bin/Release/net7.0/Jellyfin.Api.dll                                          
154.6 /repo/Emby.Server.Implementations/ApplicationHost.cs(156,38): error CS7036: There is no argument given that corresponds to the required parameter 'shortcutHandlers' of 'ManagedFileSystem.ManagedFileSystem(ILogger<ManagedFileSystem>, IApplicationPaths, IEnumerable<IShortcutHandler>)' [/repo/Emby.Server.Implementations/Emby.Server.Implementations.csproj]
154.6 /repo/Emby.Server.Implementations/ApplicationHost.cs(159,55): error CS1729: 'MbLinkShortcutHandler' does not contain a constructor that takes 1 arguments [/repo/Emby.Server.Implementations/Emby.Server.Implementations.csproj]          
154.6 /repo/Emby.Server.Implementations/ApplicationHost.cs(159,32): error CS1061: 'IFileSystem' does not contain a definition for 'AddShortcutHandler' and no accessible extension method 'AddShortcutHandler' accepting a first argument of type 'IFileSystem' could be found (are you missing a using directive or an assembly reference?) [/repo/Emby.Server.Implementations/Emby.Server.Implementations.csproj]                                                                             154.6 /repo/Emby.Server.Implementations/ApplicationHost.cs(168,40): error CS1729: 'ServerConfigurationManager' does not contain a constructor that takes 4 arguments [/repo/Emby.Server.Implementations/Emby.Server.Implementations.csproj]     ------                                                                                                                  Dockerfile:76 
```

**Comment by Shadowghost** [comment]
@satmandu merge went fine, so not sure why it failed for you.

**Comment by satmandu** [comment]
Thanks for rebasing! I don't know what happened on my end, but the rebased version of course works well.

**Comment by Shadowghost** [comment]
@satmandu @cvium bugs should be fixed. I tried a lot of combinations but apparently Windows and Linux need different handling of broadcast and multicast if you don't want to bind on all interfaces...

File: Emby.Server.Implementations/Udp/UdpServer.cs [hunk 4]
```diff
@@ -99,11 +103,12 @@ private async Task BeginReceiveAsync(CancellationToken cancellationToken)
             {
                 try
                 {
-                    var result = await _udpSocket.ReceiveFromAsync(_receiveBuffer, SocketFlags.None, _endpoint, cancellationToken).ConfigureAwait(false);
-                    var text = Encoding.UTF8.GetString(_receiveBuffer, 0, result.ReceivedBytes);
+                    var endpoint = (EndPoint)new IPEndPoint(0, 0);
+                    int bytesRead = _udpSocket.ReceiveFrom(_receiveBuffer, ref endpoint);
```
Comment (Bond-009) [inline Emby.Server.Implementations/Udp/UdpServer.cs:107]: Why not use the async function

**Review by Bond-009 — APPROVED** [review]

File: Emby.Server.Implementations/Udp/UdpServer.cs (hunk 4 above)
Comment (Shadowghost) [reply Emby.Server.Implementations/Udp/UdpServer.cs]: Changed it.

File: Emby.Server.Implementations/Net/SocketFactory.cs [hunk 5]
```diff
@@ -82,16 +86,25 @@ public Socket CreateUdpMulticastSocket(IPAddress multicastAddress, IPData bindIn
...
-                socket.SetSocketOption(SocketOptionLevel.IP, SocketOptionName.AddMembership, new MulticastOption(multicastAddress, interfaceIndex));
-                socket.Bind(new IPEndPoint(multicastAddress, localPort));
+
+                if (OperatingSystem.IsLinux())
+                {
+                    socket.Bind(new IPEndPoint(multicastAddress, localPort));
+                }
+                else
+                {
+                    // Only create socket if interface supports multicast
```
Comment (cvium) [inline Emby.Server.Implementations/Net/SocketFactory.cs:98]: You're not performing any check here?

Comment (Shadowghost) [reply Emby.Server.Implementations/Net/SocketFactory.cs:98]: Pending macOS testing this will change.

**Review by cvium — APPROVED** [review]

File: Jellyfin.Networking/Manager/NetworkManager.cs [hunk 6]
```diff
@@ -1053,5 +1109,16 @@ private bool MatchesExternalInterface(IPAddress source, out string result)
...
+
+        private void PrintNetworkInformation(NetworkConfiguration config, bool debug = true)
+        {
+            var logLevel = debug ? LogLevel.Debug : LogLevel.Information;
+            _logger.Log(logLevel, "Defined LAN addresses: {0}", _lanSubnets.Select(s => s.Prefix + "/" + s.PrefixLength));
+            _logger.Log(logLevel, "Defined LAN exclusions: {0}", _excludedSubnets.Select(s => s.Prefix + "/" + s.PrefixLength));
+            _logger.Log(logLevel, "Using LAN addresses: {0}", _lanSubnets.Where(s => !_excludedSubnets.Contains(s)).Select(s => s.Prefix + "/" + s.PrefixLength));
+            _logger.Log(logLevel, "Using bind addresses: {0}", _interfaces.OrderByDescending(x => x.AddressFamily == AddressFamily.InterNetwork).Select(x => x.Address));
+            _logger.Log(logLevel, "Remote IP filter is {0}", config.IsRemoteIPFilterBlacklist ? "Blocklist" : "Allowlist");
+            _logger.Log(logLevel, "Filter list: {0}", _remoteAddressFilter.Select(s => s.Prefix + "/" + s.PrefixLength));
```
Comment (Bond-009) [inline Jellyfin.Networking/Manager/NetworkManager.cs:1121]: This looks pretty expensive, you should place this in an if that checks if the loglevel is enabled

**Review by crobibero — APPROVED** [review]
works for me on macOS

File: Jellyfin.Networking/Manager/NetworkManager.cs (hunk 6 above)
Comment (Shadowghost) [reply Jellyfin.Networking/Manager/NetworkManager.cs:1121]: Before this was always printed on any change so in the worst case it's not worse than current.

Comment (cvium) [reply Jellyfin.Networking/Manager/NetworkManager.cs:1121]: These are all IEnumerable. The delegates are allocated but the actual cost is not incurred unless it logs it. But I agree with Bond. Better to check the level is active

**Pull request merged**